ccswbs/gus
# Infinite upper to lowercase redirect fix

# Summary of changes
As per https://dev.azure.com/uoguelph/CCS/_workitems/edit/77610 Drupal redirects from uppercase to lowercase URLs (where letter case is the only difference) create an infinite loop on Netlify. This PR fixes the problem by preventing such redirects from being built and converting all aliases to lowercase.

## Frontend
- Update `gatsby-node.js` alias function to convert all Drupal aliases to lowercase
- Update `gatsby-node.js` redirect function to skip redirects that are identical except for letter case
- Update `index.js` to convert all Drupal aliases to lowercase

## Backend
N/A (Drupal does not provide a built-in method for converting custom aliases to lowercase).

# Test Plan

- Create a page and give it a custom alias with uppercase letters
- Update the page alias to a lowercase version
- Confirm the redirect is created in Drupal
- Run the build and confirm the redirect is skipped on Netlify and only the lowercase version of the URL works
